File: cst345/week6/DatabaseSQLMusicApp/DatabaseSQLMusicApp/Track.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DatabaseSQLMusicApp
{
    
    internal class Track
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string ID { get; set; }
        public string title { get; set; }
        public int Number { get; set; }
        public string video_url { get; set; }
        public string Lyrics { get; set; }

        public Track()
        {
            ID = ObjectId.GenerateNewId().ToString();
        }
    }
}

File: cst391/week_4/musicapp/src/app/models/Track.ts
export class Track {
	public id: number = -1;
	public number: number = 0;
	public title: string = "";
	public lyrics: string = "";
	public video: string = "";

	constructor(id: number, number: number, title: string, lyrics: string, video: string) {
		this.id = id;
		this.number = number;
		this.title = title;
		this.lyrics = lyrics;
		this.video = video;
	}

	get Id(): number {
		return this.id;
	}
	set Id(id: number) {
		this.id = id;
	}

	get Number(): number {
		return this.number;
	}
	set Number(number: number) {
		this.number = number;
	}

	get Title(): string {
		return this.title;
	}
	set Title(title: string) {
		this.title = title;
	}

	get Lyrics(): string {
		return this.lyrics;
	}
	set Lyrics(lyrics: string) {
		this.lyrics = lyrics;
	}

	public get Video(): string {
		return this.video;
	}

	public set Video(value: string) {
		this.video = value;
	}
}
